apps/sync/js/adapters/bookmarks.js

```diff
@@ -112,6 +112,11 @@ var BookmarksHelper = (() => {
     localRecord.name = remoteRecord.name;
     if (!localRecord.fxsyncRecords) {
       localRecord.fxsyncRecords = {};
+      // We remember if a record had already been created locally before we got
+      // remote data for that URL, so that we know not to remove it even when
+      // the remote data is deleted. This applies only to readonly sync, and
+      // will be removed when sync becomes read-write.
+      localRecord.createdLocally = true;
```

Comment:
I don't think that would add anything. Until https://github.com/michielbdejong/gaia/blob/1221227-created-locally/apps/sync/js/adapters/bookmarks.js#L440 is fixed, that would be adding dead code, since `readonly` would always be true. After, this line would either be removed if we drop readonly support, or if we keep it, this line would only be reached in the readonly case.